### PR TITLE
Add standard Slack messages and colors

### PR DESCRIPTION
**Pull request by ddgenome**
Add standard formatted Slack messages for success, error, and
warning.  Add standard colors for each case.

### CODE REVIEW

File: src/SlackMessages.ts [hunk 1]
```diff
@@ -40,9 +57,9 @@ export function url(fullUrl: string, label?: string) {
  * When userName is provided will add readable user name.
  */
 export function user(userId: string, userName?: string) {
-    if (!emptyString(userId) && !emptyString(userName)) {
+    if (userId && userName) {
```
Comment (tanya-coding) [inline src/SlackMessages.ts]: Same here, we don't want to try to convert empty strings into `<|>` or `<>`. Same applies to channel, bold (** will render as 2 starts which is not correct), italic, etc.

File: src/SlackMessages.ts [hunk 2]
```diff
@@ -98,61 +115,37 @@ export function render(message: SlackMessage, pretty: boolean = false): string {
 
 /** Render emoji by name */
 export function emoji(name: string) {
-    return `:${name}:`;
+    return (name) ? `:${name}:` : "";
```
Comment (tanya-coding) [inline src/SlackMessages.ts]: We should add emptyString check here

File: src/SlackMessages.ts [hunk 3]
```diff
@@ -98,61 +115,37 @@ export function render(message: SlackMessage, pretty: boolean = false): string {
...
 }
 
 /** Render bold text */
 export function bold(text: string) {
-    if (!emptyString(text)) {
-        return `*${text}*`;
-    } else {
-        return "";
-    }
+    return (text) ? `*${text}*` : "";
```
Comment (tanya-coding) [inline src/SlackMessages.ts]: I think we still need emptyString check here

File: src/SlackMessages.ts [hunk 4]
```diff
@@ -98,61 +115,37 @@ export function render(message: SlackMessage, pretty: boolean = false): string {
...
 }
 
 /** Render italic text */
 export function italic(text: string) {
-    if (!emptyString(text)) {
-        return `_${text}_`;
-    } else {
-        return "";
-    }
+    return (text) ? `_${text}_` : "";
```
Comment (tanya-coding) [inline src/SlackMessages.ts]: I think we still need emptyString check here

File: src/SlackMessages.ts [hunk 5]
```diff
@@ -98,61 +115,37 @@ export function render(message: SlackMessage, pretty: boolean = false): string {
...
 }
 
 /** Render strike-through text */
 export function strikethrough(text: string) {
-    if (!emptyString(text)) {
-        return `~${text}~`;
-    } else {
-        return "";
-    }
+    return (text) ? `~${text}~` : "";
```
Comment (tanya-coding) [inline src/SlackMessages.ts]: I think we still need emptyString check here

File: src/SlackMessages.ts [hunk 6]
```diff
@@ -98,61 +115,37 @@ export function render(message: SlackMessage, pretty: boolean = false): string {
...
 }
 
 /** Render single line code block */
 export function codeLine(text: string) {
-    if (!emptyString(text)) {
-        return "`" + text + "`";
-    } else {
-        return "";
-    }
+    return (text) ? "`" + text + "`" : "";
```
Comment (tanya-coding) [inline src/SlackMessages.ts]: I think we still need emptyString check here

File: src/SlackMessages.ts [hunk 7]
```diff
@@ -98,61 +115,37 @@ export function render(message: SlackMessage, pretty: boolean = false): string {
...
 }
 
 /** Render multiline code block */
 export function codeBlock(text: string) {
-    if (!emptyString(text)) {
-        return "```" + text + "```";
-    } else {
-        return "";
-    }
+    return (text) ? "```" + text + "```" : "";
```
Comment (tanya-coding) [inline src/SlackMessages.ts]: I think we still need emptyString check here

File: src/SlackMessages.ts [hunk 8]
```diff
@@ -98,61 +115,37 @@ export function render(message: SlackMessage, pretty: boolean = false): string {
...
 }
 
 /** Render bullet list item */
 export function listItem(item: string) {
-    if (!emptyString(item)) {
-        return `• ${item}`;
-    } else {
-        return "";
-    }
+    return (item) ? `• ${item}` : "";
```
Comment (tanya-coding) [inline src/SlackMessages.ts]: I think we still need emptyString check here

File: src/SlackMessages.ts [hunk 9]
```diff
@@ -237,7 +230,7 @@ export class ValidationError extends Error {
 
 /** Construct Slack button that will execute provided rug instruction. */
 export function rugButtonFrom(action: ButtonSpec, command: IdentifiableInstruction): Action {
-    if (emptyString(command.id)) {
+    if (!command.id) {
```
Comment (tanya-coding) [inline src/SlackMessages.ts]: I don't think we should accept empty string as action ID

File: src/SlackMessages.ts [hunk 10]
```diff
@@ -253,3 +246,97 @@ export function rugButtonFrom(action: ButtonSpec, command: IdentifiableInstructi
     }
     return button;
 }
+
+interface StandardAttachmentDefaults {
```
Comment (tanya-coding) [inline src/SlackMessages.ts]: Should we move these into a separate file in order to keep slack-messages more generic?

**Comment by tanya-coding** [comment]
Added some comments. I think removing emptyString check in quite a few functions will have negative effect

**Comment by ddgenome** [comment]
@tanya-coding the empty string is false in TypeScript/JavaScript (https://javascriptweblog.wordpress.com/2011/02/07/truth-equality-and-javascript/), so you will get the proper result if the argument passed in is `null`, `undefined`, or the empty string.

Essentially, the `emptyString` test is a much more complicated version of what JavaScript does by default.

**Comment by ddgenome** [comment]
I've added SlackMessages test cases for empty string input everywhere there were tests cases for `null` and/or `undefined`. They all pass.

File: src/SlackMessages.ts (hunk 10 above)
Comment (ddgenome) [inline src/SlackMessages.ts]: I'm happy to move it to a different file. I just though because these were Slack messages, they belonged in SlackMessages.

Comment (tanya-coding) [inline src/SlackMessages.ts]: There is RugMessages.ts where something similar is already being attempted - maybe just replace/enhance what is done there with new standard messages?

**Comment by tanya-coding** [comment]
LGTM 👍

**Comment by ddgenome** [comment]
@tanya-coding I've moved the standard message stuff to RugMessages, added versions that return ResponseMessages, and deprecated renderError and renderSuccess. I deprecated rather than updated the latter two because I felt the changes were sufficiently different that I didn't want to change the message format out from people.

**Pull request merged**